### ISSUE TITLE
ProtoFlux String Split node

### ISSUE THREAD

**Issue by j4lc** [Yellow-Dog-Man/Resonite-Issues#1572]
### Is your feature request related to a problem? Please describe.

When writing systems parsing data (for instance, from CSV), it can be a bit tedious to have parsing, taking a lot of nodes for something that could probably be combined in a single one.

### Describe the solution you'd like

Add a String Split ProtoFlux node that would act like the Python `.split` by returning a collection of the split string.

For instance:
```python
my_string = "Hello,World"
print(my_string.split(","))
```
Would print out:
```
['Hello', 'World']
```

This would remove some complexity in ProtoFlux systems, though would have some pre-requisites (see additional context).

### Describe alternatives you've considered

Use the current method combining `Substring` and `IndexOfString`. Tho a bit more complex than a single node would be, it works.

### Additional Context

Would require #572 to be implemented first.

### Requesters

j4 | j4.lc (Discord)

**Comment by Frooxius** [comment]
Yeah this will very likely come after collections are added, it's a very common operation!

**Comment by copygirl** [comment]
In the meantime, it could be very useful to have a `SplitN` node that (at least) takes a string, a separator, and outputs a boolean + N number of strings. The boolean specifies whether the string could be split N times. The output strings are the separate splits.

An option to specify split behavior could be added to the node.
(Names not final just what I could come up with in the middle of the night.)
- **Exact:** String has to be split exactly a specified number of times. If more or less, the boolean will be false.  
- **Trim:** If the string has more than N elements, the remaining elements are dropped.
- **Remaining:** If the string has more than N elements, the final split will include the remaining text.

Here's an example how a `Split3` node (`SplitN` with 3 outputs) would work:

| Input | Behavior | Output |
| --- | --- | --- |
| "foo bar" | Exact | ❌, "foo", "bar", null |
| "foo bar" | Trim | ❌, "foo", "bar", null |
| "foo bar" | Remaining | ❌, "foo", "bar", null |
| "foo bar baz" | Exact | ✔️, "foo", "bar", "baz" |
| "foo bar baz" | Trim | ✔️, "foo", "bar", "baz" |
| "foo bar baz" | Remaining | ✔️, "foo", "bar", "baz" |
| "foo bar baz quux" | Exact | ❌, "foo", "bar", "baz" |
| "foo bar baz quux" | Trim | ✔️, "foo", "bar", "baz" |
| "foo bar baz quux" | Remaining | ✔️, "foo", "bar", "baz quux" |

This node could still be useful even if collection support and a `Split` node were added, since it combines a bunch of extra functionality such as number-of-splits checking, individual item extraction and string re-combination. (Though `Split` should have a "max splits" option so that last part might not be that big of a deal.)

**Comment by Frooxius** [comment]
This is now added in 2026.8.26.1047!
